packages/web/src/pages/artist-dashboard-page/store/sagas.ts

```diff
@@ -0,0 +1,204 @@
+import {
+  IntKeys,
+  accountSelectors,
+  walletActions,
+  doEvery,
+  waitForValue,
+  ID,
+  getContext,
+  Collection,
+  Track
+} from '@audius/common'
+import { each } from 'lodash'
+import moment from 'moment'
+import { all, call, put, take, takeEvery } from 'typed-redux-saga'
+
+import { retrieveUserTracks } from 'common/store/pages/profile/lineups/tracks/retrieveUserTracks'
+import { requiresAccount } from 'common/utils/requiresAccount'
+import { DASHBOARD_PAGE } from 'utils/route'
+import { waitForRead } from 'utils/sagaHelpers'
+
+import { actions as dashboardActions } from './slice'
+import ArtistDashboardState from './types'
+const { getBalance } = walletActions
+const getAccountUser = accountSelectors.getAccountUser
+
+const formatMonth = (date: moment.Moment | string) =>
+  moment.utc(date).format('MMM').toUpperCase()
+
+function* fetchDashboardTracksAsync(
+  action: ReturnType<typeof dashboardActions.fetchTracks>
+) {
+  const account = yield* call(waitForValue, getAccountUser)
+  const { offset, limit } = action.payload
+
+  try {
+    const tracks = yield* call(retrieveUserTracks, {
+      handle: account.handle,
+      currentUserId: account.user_id,
+      offset,
+      limit,
+      getUnlisted: true
+    })
+    yield put(dashboardActions.fetchTracksSucceeded({ tracks }))
+  } catch (error) {
+    console.error(error)
+    yield put(dashboardActions.fetchTracksFailed({}))
+  }
+}
+
+function* fetchDashboardAsync(
+  action: ReturnType<typeof dashboardActions.fetch>
+) {
+  const audiusBackendInstance = yield* getContext('audiusBackendInstance')
+  yield* call(waitForRead)
+
+  const account = yield* call(waitForValue, getAccountUser)
+  const { offset, limit } = action.payload
+
+  try {
+    const data = yield* all([
+      call(retrieveUserTracks, {
+        handle: account.handle,
+        currentUserId: account.user_id,
+        offset,
+        limit,
+        getUnlisted: true
+      }),
+      call(audiusBackendInstance.getPlaylists, account.user_id, [])
+    ])
+    // Casting necessary because yield* all is not typed well
+    const tracks = data[0] as Track[]
+    const playlists = data[1] as Collection[]
+
+    const trackIds = tracks.map((t) => t.track_id)
+    const now = moment()
+
+    yield* put(
+      dashboardActions.fetchListenData({
+        trackIds,
+        start: now.clone().subtract(1, 'years').toISOString(),
+        end: now.toISOString(),
+        period: 'month'
+      })
+    )
+
+    if (tracks.length > 0 || playlists.length > 0) {
+      yield* put(
+        dashboardActions.fetchSucceeded({
+          tracks,
+          collections: playlists
+        })
+      )
+      yield* call(pollForBalance)
```

Comment:
curious why we're only calling this after fetching tracks instead of concurrently along with fetching tracks?

Reply:
good call. no idea. this was copied from before, but i'll change to fork before